controllers/ppwebhook.go

```diff
@@ -0,0 +1,470 @@
+package controllers
+
+import (
+	"context"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	// import apps/v1 for Deployment
+	appsv1 "k8s.io/api/apps/v1"
+	// import metav1 for ObjectMeta
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// import admissionregistration/v1 for MutatingWebhookConfiguration
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+)
+
+const (
+
+	// Define webhook name
+	webhookName = "mwebhook.peerpods.io"
+
+	// Define webhook deployment name
+	webhookDeploymentName = "peer-pods-webhook"
+
+	// Define webhook service name
+	webhookSvcName       = "peer-pods-webhook-svc"
+	webhookSvcSecretName = "peer-pods-webhook-svc-secret"
+	webhookDefaultImage  = "quay.io/confidential-containers/peer-pods-webhook:latest"
+
+	// Define webhook mutating config name
+	webhookConfigName = "mutating-webhook-configuration"
+)
+
+// Method to create the mutating webhook service
+func (r *KataConfigOpenShiftReconciler) createMutatingWebhookService() error {
+
+	// Define webhook service port
+	webhookServicePort := int32(443)
+
+	// Define webhook service target port
+	webhookServiceTargetPort := intstr.FromInt(9443)
+
+	// Define webhook service type
+	webhookServiceType := corev1.ServiceTypeClusterIP
+
+	// Define webhook service selector
+	webhookServiceSelector := map[string]string{
+		"app": "peer-pods-webhook",
+	}
+
+	webhookSvcNamespace := os.Getenv("PEERPODS_NAMESPACE")
+
+	// Define Annotation for webhook service
+	webhookServiceAnnotations := map[string]string{
+		// annotation to the service to use the secret created by the operator
+		// to serve the webhook service over TLS
+		"service.beta.openshift.io/serving-cert-secret-name": webhookSvcSecretName,
+	}
+
+	// Create webhook service
+	webhookService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      webhookSvcName,
+			Namespace: webhookSvcNamespace,
+			// Add annotations to the service
+			Annotations: webhookServiceAnnotations,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       webhookSvcName,
+					Port:       webhookServicePort,
+					TargetPort: webhookServiceTargetPort,
+				},
+			},
+			Selector:  webhookServiceSelector,
+			ClusterIP: "",
+			Type:      webhookServiceType,
+		},
+	}
+
+	// Create webhook service
+	if err := r.Client.Create(context.Background(), webhookService); err != nil {
+		// Check if the webhook service already exists
+		if !k8serrors.IsAlreadyExists(err) {
+			r.Log.Error(err, "peerpods mutating webhook service already exists")
```

Comment:
The error message is backwards with respect to the check. Since the error is already logged by the caller, what about just dropping this line ?